### DISCUSSION TITLE
ultrasonic/distance sensor use

### DISCUSSION THREAD

**Discussion by incredibelles** [pybricks/support#1932] [Q&A]
I'm having trouble wrapping my head around the ultrasonic sensor use in pybricks and could use a little help. I've made it this far:
<img width="754" alt="image" src="https://github.com/user-attachments/assets/1749ac0b-9a16-4327-9855-d8d7f3d9f62a">

but this prints "UltrasonicSensor"

<img width="312" alt="image" src="https://github.com/user-attachments/assets/8f9da849-2661-4c05-9061-6c988aadbe8a">

and this prints "0"

So how do I retrieve the current ultrasonic/distance sensor value?
Ultimately, I'm trying to get to something like this (I think? does this code look right? i can't test it as I don't know the value I'll need)
<img width="483" alt="image" src="https://github.com/user-attachments/assets/f56659ee-cc52-4b47-b6e0-0a8e3af46bfe">
where the robot is checking the sensor data and will drive until the distance is achieved, stop, then turn left.

**Comment by BertLindeman** [comment]
Hello,

Do you need these blocks?
![pybricks_program_2024-11-13T14_43_41](https://github.com/user-attachments/assets/17f3de1a-2abc-4497-8cc1-d7620dcbedbe)

The distance block is in the input palette.

Bert

**Reply by BertLindeman** [reply]
Maybe an extra pointer about the block colors.

In the documentation on the pybricks site all relative blocks are shown.
[EDIT] With the exception of setup blocks and the comments block.[/EDIT]
The color of the blocks indicate in which palette they can be found.

As example the ultra-sonic sensor:
![image](https://github.com/user-attachments/assets/f4df601e-9555-4c4f-8fb3-f497fc04ba95)

The same color as the input palette has.
![image](https://github.com/user-attachments/assets/ea14e8a4-09f6-43ef-b327-3a4e174953a2)

Hope this helps a bit.

Bert

**Reply by incredibelles** [reply]
Thanks, Bert! Can you provide an example of where I might use the variable (pink) instead?

**Reply by BertLindeman** [reply]
Sure 😃 

Define a variable and fill it like this maybe:
![pybricks_program_2024-11-14T14_35_36](https://github.com/user-attachments/assets/55fbd3b7-35cc-4e82-bfca-808c9b54b196)
